### PR TITLE
Editor reviews and edits article

### PR DESCRIPTION
**Pull request by dorianbuck**
PT: https://www.pivotaltracker.com/story/show/180061274

## UserStory
```
As an editor
In order to ensure high quality articles
I would like to be able to review and edit articles
```

## Tasks
- Create and scaffold an `EditArticle` component
- Create a Get request for a specific article
- Store the categories in application state when initially getting all articles
- Create a Put request to update edited article
- Redirect user to `review-articles` page after submitting edits

### ScreenShots
eZW9DgClmG

image
image

image

### CODE REVIEW

**Review by dorianbuck — COMMENTED** [review]
There seems to be some formating issues on this PR. I do take some responsibility, we all have to be more mindful of this. Well done. It looks great!

File: src/components/EditArticle.jsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+    const editedArticle = {
+      article: {
+        id: id,
+        title: title,
+        authors: authors,
+        lede: lede,
+        category: category,
+        body: body,
+      },
```
Comment (lukeperrin10) [inline src/components/EditArticle.jsx]: can't something like this work better?  few less lines of code too

```js

const editedArticle = {
   article: {
        id: id,
        title: event.target.title.value
        authors: [event.target.title.journalist];
        lede: event.target.lede.value;
        category: selectedCategory;
        body: event.target.body.value;
      }       
}

```
or 

```js
const form = event.target
const editedArticle = {
   article: {
        id: id,
        title: form.title.value
        authors: [form.title.journalist];
        lede: form.lede.value;
        category: selectedCategory;
        body: form.body.value;
      }       
}

```

File: cypress/integration/userCanEditAnArticle.feature.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      cy.get("span").contains("Business").click();
+    });
+    cy.get("[data-cy=submit-btn]").click();
+    cy.get("[data-cy=response-message]").should(
+      "contain.text",
+      "You have successfully edited the article"
+    );
+  });
+
+  it.only('is expected to redirect to the list of articles after submission', () => {
```
Comment (Fraseer) [inline cypress/integration/userCanEditAnArticle.feature.js]: Better remove this

**Review by Fraseer — CHANGES_REQUESTED** [review]

File: src/components/EditArticle.jsx (hunk 1 above)
Comment (Fraseer) [reply src/components/EditArticle.jsx]: Reformatted to the better version

File: cypress/integration/userCanEditAnArticle.feature.js (hunk 2 above)
Comment (Fraseer) [reply cypress/integration/userCanEditAnArticle.feature.js]: Removed

**Review by Fraseer — APPROVED** [review]

**Review by lukeperrin10 — APPROVED** [review]
Good work

**Pull request merged**